OpenRA.Mods.Common/Traits/Render/ProductionBar.cs

```diff
@@ -16,14 +16,30 @@
 namespace OpenRA.Mods.Common.Traits.Render
 {
 	[Desc("Visualizes the remaining build time of actor produced here.")]
-	class ProductionBarInfo : ConditionalTraitInfo, Requires<ProductionInfo>
+	class ProductionBarInfo : ConditionalTraitInfo, Requires<ProductionInfo>, IRulesetLoaded
 	{
 		[FieldLoader.Require]
 		[Desc("Production queue type, for actors with multiple queues.")]
 		public readonly string ProductionType = null;
 
 		public readonly Color Color = Color.SkyBlue;
 
+		public override void RulesetLoaded(Ruleset rules, ActorInfo ai)
+		{
+			// Per-actor queue
+			var queue = ai.TraitInfos<ProductionQueueInfo>().FirstOrDefault(q => ProductionType == q.Type);
+			if (queue == null)
+			{
+				// No queues available - check for classic production queues
```

Comment:
If you move the comment before the `if`, you can drop the brackets.

Reply:
Ping @Mailaender